features/build/build_websocket_backend.rb

```diff
@@ -53,22 +53,17 @@ def call(env)
         self.websocket_clients[project_id][build_number] ||= []
         self.websocket_clients[project_id][build_number] << ws
 
-        BuildRunnerService.build_runner_services.each do |build_runner_service|
-          next if build_runner_service.current_build.number != build_number
-          next if build_runner_service.project.id != project_id
-
-          # TODO: Think this through, do we properly add new listener, and notify them of line changes, etc.
-          #       Also how does the "offboarding" of runners work once the tests are finished
-          build_runner_service.add_listener(proc do |row|
-            web_sockets = self.websocket_clients[project_id][build_number]
-            logger.debug("Streaming #{row} to #{web_sockets.count} client(s)")
-
-            web_sockets.each do |current_socket|
-              # TODO: Add auth check here, so a user isn't able to get the log from another build
-              current_socket.send(row.to_json)
-            end
-          end)
-        end
+        current_build_runner = Services.build_runner_service.find_build_runner(
+          project_id: project_id,
+          build_number: build_number
+        )
+
+        # TODO: Think this through, do we properly add new listener, and notify them of line changes, etc.
+        #       Also how does the "offboarding" of runners work once the tests are finished
+        current_build_runner.add_listener(proc do |row|
```

Comment:
I think it's clearer as
`current_build_runner.add_listener(proc { |row| ws.send(row.to_json)})` or whatevs the syntax is if this isn't right

Reply:
That's what I have so far. Or do you refer to the one line approach vs multi-line? The reason for multi-line is that I need to add an additional auth check probably here, to avoid people accessing logs from builds they don't have access to

Reply:
I was referring to the one-line approach if you have just 1 single statement. Since you're going to add stuff, you can ignore me here 👍